fg-stitch-lib/src/align/aligners/mod.rs

```diff
@@ -838,29 +910,33 @@ impl<'a, F: MatchFunc> SamRecordFormatter<'a, F> {
                 );
                 data.insert(
                     "as".parse().unwrap(),
-                    noodles::sam::record::data::field::Value::from(alignment.score),
+                    noodles::sam::record::data::field::Value::from(chain.score),
                 );
-                if let Some(score) = alt_score {
+                if let Some(score) = suboptimal_score {
                     data.insert(
                         "xs".parse().unwrap(),
                         noodles::sam::record::data::field::Value::from(score),
                     );
                 }
                 data.insert(
-                    ALIGNMENT_SCORE,
-                    noodles::sam::record::data::field::Value::from(sub.score),
+                    "si".parse().unwrap(),
+                    noodles::sam::record::data::field::Value::from(sub_index as i32),
                 );
                 data.insert(
-                    HIT_INDEX,
-                    noodles::sam::record::data::field::Value::from(index as i32 + 1),
+                    "cl".parse().unwrap(),
+                    noodles::sam::record::data::field::Value::from(subs.len() as i32),
                 );
                 data.insert(
-                    ALIGNMENT_HIT_COUNT,
-                    noodles::sam::record::data::field::Value::from(subs.len() as i32),
+                    "ci".parse().unwrap(),
+                    noodles::sam::record::data::field::Value::from(chain_index as i32),
                 );
                 data.insert(
-                    TOTAL_HIT_COUNT,
-                    noodles::sam::record::data::field::Value::from(subs.len() as i32),
+                    "cn".parse().unwrap(),
+                    noodles::sam::record::data::field::Value::from(chains.len() as i32),
+                );
+                data.insert(
+                    ALIGNMENT_SCORE,
+                    noodles::sam::record::data::field::Value::from(sub.score),
```

Comment:
I think we should add some unit tests for these tags.  Perhaps we spot check some alignments manually, then create an issue to add tests!